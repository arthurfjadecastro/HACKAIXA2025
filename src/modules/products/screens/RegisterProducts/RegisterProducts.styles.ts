import { StyleSheet } from 'react-native';
import { colors, spacing } from '@/design-system/tokens';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.background.primary,
  },
  
  // Header
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: spacing[4],
    paddingTop: spacing[8],
    paddingBottom: spacing[4],
    backgroundColor: colors.background.primary,
  },
  backButton: {
    padding: spacing[2],
    marginRight: spacing[3],
  },
  headerTitle: {
    flex: 1,
    color: colors.primary.main,
    fontSize: 20,
    fontWeight: '600',
  },
  addButton: {
    padding: spacing[2],
    marginLeft: spacing[3],
  },
  
  // Content
  scrollView: {
    flex: 1,
    paddingHorizontal: spacing[4],
  },
  
  // Empty state
  emptyContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: spacing[12],
  },
  emptyTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: colors.text.secondary,
    marginTop: spacing[4],
    marginBottom: spacing[2],
  },
  emptySubtitle: {
    fontSize: 14,
    color: colors.text.secondary,
    textAlign: 'center',
    paddingHorizontal: spacing[6],
  },
  
  // Products
  productsContainer: {
    paddingVertical: spacing[4],
    gap: spacing[4], // Adiciona espaçamento vertical entre os cards
  },
  
  // Botão Criar Novo Produto
  createNewButton: {
    borderWidth: 2,
    borderColor: colors.primary.main,
    borderStyle: 'dashed',
    borderRadius: 8,
    paddingVertical: spacing[4],
    paddingHorizontal: spacing[4],
    marginBottom: spacing[6],
    backgroundColor: 'transparent',
  },
  createNewContent: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  createNewText: {
    marginLeft: spacing[2],
    color: colors.primary.main,
    fontSize: 16,
    fontWeight: '500',
  },
  
  // Lista de Produtos
  productsList: {
    gap: spacing[4],
  },
  productCard: {
    backgroundColor: colors.background.secondary,
    borderRadius: 12,
    borderWidth: 1,
    borderColor: colors.border.light,
    padding: spacing[4],
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.1,
    shadowRadius: 2,
    elevation: 2,
  },
  productCardSelected: {
    borderColor: colors.primary.main,
    borderWidth: 2,
    backgroundColor: '#F0F7FF', // Fundo levemente azul quando selecionado
  },
  
  // Header do Card
  productHeader: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: spacing[3],
  },
  productIconContainer: {
    width: 32,
    height: 32,
    borderRadius: 6,
    backgroundColor: colors.background.primary,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: spacing[3],
  },
  productTitle: {
    fontSize: 16,
    fontWeight: '600',
    color: colors.text.primary,
  },
  
  // Detalhes do Produto
  productDetails: {
    gap: spacing[2],
  },
  detailRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  detailLabel: {
    fontSize: 14,
    color: colors.text.secondary,
  },
  detailValue: {
    fontSize: 14,
    color: colors.text.primary,
    fontWeight: '500',
  },
  
  // Footer
  footer: {
    padding: spacing[4],
    backgroundColor: '#FFFFFF', // Fundo branco para melhor contraste
    borderTopWidth: 1,
    borderTopColor: '#E0E0E0', // Borda mais visível
    elevation: 8, // Sombra para destacar
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: -2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 3.84,
  },
  saveButton: {
    width: '100%',
    backgroundColor: '#F59E0B', // Laranja CAIXA para melhor visibilidade
    borderRadius: 12,
    elevation: 4, // Sombra no Android
    shadowColor: '#000', // Sombra no iOS
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 3.84,
  },
  
  // Legacy (manter para compatibilidade)
  centerContent: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingHorizontal: spacing[4],
  },
  centerText: {
    textAlign: 'center',
  },
  
  // Badge de Status
  statusBadge: {
    paddingHorizontal: spacing[2],
    paddingVertical: spacing[1],
    borderRadius: 8,
    minWidth: 60,
    alignItems: 'center',
  },
  activeBadge: {
    backgroundColor: '#E8F5E8',
  },
  inactiveBadge: {
    backgroundColor: '#FFF3E0',
  },
  statusText: {
    fontSize: 12,
    fontWeight: '600',
  },
  activeText: {
    color: '#2E7D32',
  },
  inactiveText: {
    color: '#F57C00',
  },
  
  // Status badge
  statusBadgeActive: {
    backgroundColor: '#E8F5E8',
  },
  statusBadgeInactive: {
    backgroundColor: '#FFF3E0',
  },
  statusTextActive: {
    color: '#2E7D32',
  },
  statusTextInactive: {
    color: '#F57C00',
  },
  
  // Produto info
  productInfo: {
    flex: 1,
    flexDirection: 'row', // Organiza badge e conteúdo em linha
    alignItems: 'center', // Alinha verticalmente
    justifyContent: 'space-between', // Distribui espaço
  },
  productTitleRow: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 4, // Margem reduzida
    flex: 1, // Ocupa espaço disponível
  },
  productIcon: {
    marginRight: spacing[2],
  },
  productName: {
    fontSize: 16,
    fontWeight: '600',
    color: colors.text.primary,
    flex: 1,
  },
  
  // Row para badge e botão de deletar
  actionRow: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 8, // Espaçamento entre badge e botão
  },
  
  // Selection indicator
  selectionIndicator: {
    position: 'absolute',
    top: spacing[2],
    right: spacing[2],
  },
  
  // Botão de deletar
  deleteButton: {
    marginLeft: 8, // Margem reduzida
    padding: 6, // Padding menor para ficar mais sutil
    borderRadius: 12, // Cantos mais suaves
    backgroundColor: 'rgba(220, 38, 38, 0.08)', // Fundo vermelho muito sutil
    alignItems: 'center',
    justifyContent: 'center',
    width: 28, // Largura fixa
    height: 28, // Altura fixa para melhor alinhamento
  },
});
