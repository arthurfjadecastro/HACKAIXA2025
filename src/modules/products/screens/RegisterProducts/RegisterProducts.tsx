import React, { useState } from 'react';
import { View, ScrollView, TouchableOpacity, Alert } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';

import { Text, Button } from '@/design-system/components';
import { AppStackParamList } from '@/navigation/AppStack';
import { styles } from './RegisterProducts.styles';
import { useProductManagement } from '@/modules/products/hooks/useProductManagement';
import { useDeleteProduct } from '@/hooks/useDeleteProduct';

type NavigationProps = NativeStackNavigationProp<AppStackParamList>;

const RegisterProducts: React.FC = () => {
  const navigation = useNavigation<NavigationProps>();
  const { products, toggleProductStatus } = useProductManagement();
  const { deleteProduct } = useDeleteProduct();

  // Estado para controlar seleções dos produtos cadastrados
  const [selectedProducts, setSelectedProducts] = useState<string[]>([]);
  
  // Lista de produtos cadastrados (sem template)
  const allItems = products.map(p => ({
    ...p,
    type: 'product' as const,
    icon: 'briefcase-outline',
    isTemplate: false,
    selected: selectedProducts.includes(p.id)
  }));

  const handleGoBack = () => {
    navigation.goBack();
  };

  const handleToggleItem = (itemId: string) => {
    setSelectedProducts(prev => 
      prev.includes(itemId)
        ? prev.filter(id => id !== itemId)
        : [...prev, itemId]
    );
  };

  const handleSaveProducts = async () => {
    // Processar produtos cadastrados selecionados
    for (const productId of selectedProducts) {
      try {
        await toggleProductStatus(productId);
      } catch (error) {
        console.error('Erro ao processar produto:', error);
      }
    }
    
    // Limpar seleções de produtos
    setSelectedProducts([]);
  };

  const handleCreateNewProduct = () => {
    // Navegar para o formulário de criação de produto
    navigation.navigate('CreateProduct');
  };

  const selectedCount = selectedProducts.length;

  return (
    <View style={styles.container}>
      {/* Header */}
      <View style={styles.header}>
        <TouchableOpacity 
          onPress={handleGoBack}
          style={styles.backButton}
          testID="back-button"
        >
          <Ionicons name="arrow-back" size={24} color="#195C5B" />
        </TouchableOpacity>
        
        <Text style={styles.headerTitle}>Produtos</Text>
        
        <TouchableOpacity 
          onPress={handleCreateNewProduct}
          style={styles.addButton}
          testID="add-product-button"
        >
          <Ionicons name="add" size={24} color="#195C5B" />
        </TouchableOpacity>
      </View>

      {/* Lista de produtos */}
      <ScrollView style={styles.scrollView} showsVerticalScrollIndicator={false}>
        {allItems.length === 0 ? (
          <View style={styles.emptyContainer}>
            <Ionicons name="folder-open-outline" size={64} color="#B0B0B0" />
            <Text style={styles.emptyTitle}>Nenhum produto cadastrado</Text>
            <Text style={styles.emptySubtitle}>
              Clique no botão + para criar seu primeiro produto
            </Text>
          </View>
        ) : (
          <View style={styles.productsContainer}>
            {allItems.map((item) => (
              <TouchableOpacity
                key={item.id}
                style={[
                  styles.productCard,
                  item.selected && styles.productCardSelected
                ]}
                onPress={() => handleToggleItem(item.id)}
                testID={`product-item-${item.id}`}
              >
                <View style={styles.productHeader}>
                  <View style={styles.productInfo}>
                    <View style={styles.productTitleRow}>
                      <Ionicons 
                        name={item.icon as any} 
                        size={20} 
                        color="#195C5B" 
                        style={styles.productIcon}
                      />
                      <Text style={styles.productName}>{item.name}</Text>
                    </View>
                    
                    <View style={[
                      styles.statusBadge,
                      item.active ? styles.statusBadgeActive : styles.statusBadgeInactive
                    ]}>
                      <Text style={[
                        styles.statusText,
                        item.active ? styles.statusTextActive : styles.statusTextInactive
                      ] as any}>
                        {item.active ? 'ATIVO' : 'INATIVO'}
                      </Text>
                    </View>
                  </View>
                  
                  {/* Botão de deletar */}
                  <TouchableOpacity
                    style={styles.deleteButton}
                    onPress={() => {
                      Alert.alert(
                        'Excluir produto',
                        `Tem certeza que deseja excluir "${item.name}"?`,
                        [
                          { text: 'Cancelar', style: 'cancel' },
                          { 
                            text: 'Excluir', 
                            style: 'destructive', 
                            onPress: () => deleteProduct(item.id) 
                          }
                        ]
                      );
                    }}
                    hitSlop={{ top: 8, bottom: 8, left: 8, right: 8 }}
                  >
                    <Ionicons name="trash-outline" size={20} color="#F44336" />
                  </TouchableOpacity>
                </View>
                
                <View style={styles.productDetails}>
                  <View style={styles.detailRow}>
                    <Text style={styles.detailLabel}>Juros</Text>
                    <Text style={styles.detailValue}>{item.juros}% a.a.</Text>
                  </View>
                  <View style={styles.detailRow}>
                    <Text style={styles.detailLabel}>Prazo máximo</Text>
                    <Text style={styles.detailValue}>{item.prazoMaximo} meses</Text>
                  </View>
                  <View style={styles.detailRow}>
                    <Text style={styles.detailLabel}>Normativo</Text>
                    <Text style={styles.detailValue}>{item.normativo}</Text>
                  </View>
                </View>
                
                {/* Indicador de seleção */}
                {item.selected && (
                  <View style={styles.selectionIndicator}>
                    <Ionicons name="checkmark-circle" size={24} color="#4CAF50" />
                  </View>
                )}
              </TouchableOpacity>
            ))}
          </View>
        )}
      </ScrollView>

      {/* Footer com botão de salvar */}
      {selectedCount > 0 && (
        <View style={styles.footer}>
          <Button
            onPress={handleSaveProducts}
            variant="primary"
            style={styles.saveButton}
            testID="save-button"
            title={`Salvar alterações (${selectedCount})`}
          />
        </View>
      )}
    </View>
  );
};

export default RegisterProducts;
