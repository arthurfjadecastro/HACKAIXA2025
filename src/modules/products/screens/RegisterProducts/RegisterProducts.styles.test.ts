import { styles } from './RegisterProducts.styles';

describe('RegisterProducts Styles', () => {
  it('should export styles object', () => {
    expect(styles).toBeDefined();
    expect(typeof styles).toBe('object');
  });

  it('should have container style', () => {
    expect(styles.container).toBeDefined();
    expect(styles.container.flex).toBe(1);
  });

  it('should have header style', () => {
    expect(styles.header).toBeDefined();
    expect(styles.header.flexDirection).toBe('row');
    expect(styles.header.alignItems).toBe('center');
  });

  it('should have backButton style', () => {
    expect(styles.backButton).toBeDefined();
    expect(styles.backButton.padding).toBe(8);
    expect(styles.backButton.marginLeft).toBe(-8);
  });

  it('should have headerTitle style', () => {
    expect(styles.headerTitle).toBeDefined();
    expect(styles.headerTitle.flex).toBe(1);
    expect(styles.headerTitle.fontSize).toBe(20);
    expect(styles.headerTitle.fontWeight).toBe('600');
  });

  it('should have addButton style', () => {
    expect(styles.addButton).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(styles).toMatchSnapshot();
  });
});
