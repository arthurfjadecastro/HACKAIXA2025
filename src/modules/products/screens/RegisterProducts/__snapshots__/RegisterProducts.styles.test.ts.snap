// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterProducts Styles should match snapshot 1`] = `
{
  "actionRow": {
    "alignItems": "center",
    "flexDirection": "row",
    "gap": 8,
  },
  "activeBadge": {
    "backgroundColor": "#E8F5E8",
  },
  "activeText": {
    "color": "#2E7D32",
  },
  "addButton": {
    "marginLeft": 12,
    "padding": 8,
  },
  "backButton": {
    "marginLeft": -8,
    "marginRight": 12,
    "padding": 8,
  },
  "centerContent": {
    "alignItems": "center",
    "flex": 1,
    "justifyContent": "center",
    "paddingHorizontal": 16,
  },
  "centerText": {
    "textAlign": "center",
  },
  "container": {
    "backgroundColor": "#F4F8FB",
    "flex": 1,
  },
  "createNewButton": {
    "backgroundColor": "transparent",
    "borderColor": "#005CA9",
    "borderRadius": 8,
    "borderStyle": "dashed",
    "borderWidth": 2,
    "marginBottom": 24,
    "paddingHorizontal": 16,
    "paddingVertical": 16,
  },
  "createNewContent": {
    "alignItems": "center",
    "flexDirection": "row",
    "justifyContent": "center",
  },
  "createNewText": {
    "color": "#005CA9",
    "fontSize": 16,
    "fontWeight": "500",
    "marginLeft": 8,
  },
  "deleteButton": {
    "alignItems": "center",
    "backgroundColor": "rgba(220, 38, 38, 0.08)",
    "borderRadius": 12,
    "height": 28,
    "justifyContent": "center",
    "marginLeft": 8,
    "padding": 6,
    "width": 28,
  },
  "detailLabel": {
    "color": "#666666",
    "fontSize": 14,
  },
  "detailRow": {
    "alignItems": "center",
    "flexDirection": "row",
    "justifyContent": "space-between",
  },
  "detailValue": {
    "color": "#12161C",
    "fontSize": 14,
    "fontWeight": "500",
  },
  "emptyContainer": {
    "alignItems": "center",
    "flex": 1,
    "justifyContent": "center",
    "paddingVertical": 48,
  },
  "emptySubtitle": {
    "color": "#666666",
    "fontSize": 14,
    "paddingHorizontal": 24,
    "textAlign": "center",
  },
  "emptyTitle": {
    "color": "#666666",
    "fontSize": 18,
    "fontWeight": "600",
    "marginBottom": 8,
    "marginTop": 16,
  },
  "footer": {
    "backgroundColor": "#FFFFFF",
    "borderTopColor": "#E0E0E0",
    "borderTopWidth": 1,
    "elevation": 8,
    "padding": 16,
    "shadowColor": "#000",
    "shadowOffset": {
      "height": -2,
      "width": 0,
    },
    "shadowOpacity": 0.1,
    "shadowRadius": 3.84,
  },
  "header": {
    "alignItems": "center",
    "backgroundColor": "#F4F8FB",
    "flexDirection": "row",
    "paddingBottom": 16,
    "paddingHorizontal": 16,
    "paddingTop": 32,
  },
  "headerTitle": {
    "color": "#005CA9",
    "flex": 1,
    "fontSize": 20,
    "fontWeight": "600",
  },
  "inactiveBadge": {
    "backgroundColor": "#FFF3E0",
  },
  "inactiveText": {
    "color": "#F57C00",
  },
  "productCard": {
    "backgroundColor": "#FFFFFF",
    "borderColor": "#EEEEEE",
    "borderRadius": 12,
    "borderWidth": 1,
    "elevation": 2,
    "padding": 16,
    "shadowColor": "#000",
    "shadowOffset": {
      "height": 1,
      "width": 0,
    },
    "shadowOpacity": 0.1,
    "shadowRadius": 2,
  },
  "productCardSelected": {
    "backgroundColor": "#F0F7FF",
    "borderColor": "#005CA9",
    "borderWidth": 2,
  },
  "productDetails": {
    "gap": 8,
  },
  "productHeader": {
    "alignItems": "center",
    "flexDirection": "row",
    "justifyContent": "space-between",
    "marginBottom": 12,
  },
  "productIcon": {
    "marginRight": 8,
  },
  "productIconContainer": {
    "alignItems": "center",
    "backgroundColor": "#F4F8FB",
    "borderRadius": 6,
    "height": 32,
    "justifyContent": "center",
    "marginRight": 12,
    "width": 32,
  },
  "productInfo": {
    "alignItems": "center",
    "flex": 1,
    "flexDirection": "row",
    "justifyContent": "space-between",
  },
  "productName": {
    "color": "#12161C",
    "flex": 1,
    "fontSize": 16,
    "fontWeight": "600",
  },
  "productTitle": {
    "color": "#12161C",
    "fontSize": 16,
    "fontWeight": "600",
  },
  "productTitleRow": {
    "alignItems": "center",
    "flex": 1,
    "flexDirection": "row",
    "marginBottom": 4,
  },
  "productsContainer": {
    "gap": 16,
    "paddingVertical": 16,
  },
  "productsList": {
    "gap": 16,
  },
  "saveButton": {
    "backgroundColor": "#F59E0B",
    "borderRadius": 12,
    "elevation": 4,
    "shadowColor": "#000",
    "shadowOffset": {
      "height": 2,
      "width": 0,
    },
    "shadowOpacity": 0.1,
    "shadowRadius": 3.84,
    "width": "100%",
  },
  "scrollView": {
    "flex": 1,
    "paddingHorizontal": 16,
  },
  "selectionIndicator": {
    "position": "absolute",
    "right": 8,
    "top": 8,
  },
  "statusBadge": {
    "alignItems": "center",
    "borderRadius": 8,
    "minWidth": 60,
    "paddingHorizontal": 8,
    "paddingVertical": 4,
  },
  "statusBadgeActive": {
    "backgroundColor": "#E8F5E8",
  },
  "statusBadgeInactive": {
    "backgroundColor": "#FFF3E0",
  },
  "statusText": {
    "fontSize": 12,
    "fontWeight": "600",
  },
  "statusTextActive": {
    "color": "#2E7D32",
  },
  "statusTextInactive": {
    "color": "#F57C00",
  },
}
`;
